section11/numberformat_proposed.html

```diff
@@ -202,18 +225,52 @@ <h1>PartitionNumberPattern ( _numberFormat_, _x_ )</h1>
           1. Else if _p_ is equal `"minusSign"`, then
             1. Let _minusSignSymbol_ be the ILND String representing the minus sign.
             1. Append a new Record { [[Type]]: `"minusSign"`, [[Value]]: _minusSignSymbol_ } as the last element of _result_.
-          1. Else if _p_ is equal `"percentSign"` and _numberFormat_.[[Style]] is `"percent"`, then
+          1. Else if _p_ is equal `"compactSymbol"`, then
+            1. Let _compactSymbol_ be an ILD string representing _exponent_ in short form, which may depend on _x_ in languages having different plural forms. The implementation must be able to provide this string, or else the pattern would not have a `"{compactSymbol}"` placeholder.
+            1. Append a new Record { [[Type]]: `"compact"`, [[Value]]: _compactSymbol_ } as the last element of _result_.
+          1. Else if _p_ is equal `"compactName"`, then
+            1. Let _compactName_ be an ILD string representing _exponent_ in long form, which may depend on _x_ in languages having different plural forms. The implementation must be able to provide this string, or else the pattern would not have a `"{compactName}"` placeholder.
+            1. Append a new Record { [[Type]]: `"compact"`, [[Value]]: _compactName_ } as the last element of _result_.
+          1. Else if _p_ is equal `"scientificSeparator"`, then
+            1. Let _scientificSeparator_ be the ILND String representing the exponent separator.
+            1. Append a new Record { [[Type]]: `"exponentSeparator"`, [[Value]]: _scientificSeparator_ } as the last element of _result_.
+          1. Else if _p_ is equal `"scientificExponent"`, then
+            1. If _exponent_ &lt; 0, then
+              1. Let _minusSignSymbol_ be the ILND String representing the minus sign.
+              1. Append a new Record { [[Type]]: `"exponentMinusSign"`, [[Value]]: _minusSignSymbol_ } as the last element of _result_.
+              1. Let _exponent_ be -_exponent_.
+            1. Let _exponentResult_ be ToRawFixed(_exponent_, 1, 0, 0).
+            1. Append a new Record { [[Type]]: `"exponentInteger"`, [[Value]]: _exponentResult_.[[FormattedString]] } as the last element of _result_.
+          1. Else if _p_ is equal `"percentSign"`, then
             1. Let _percentSignSymbol_ be the ILND String representing the percent sign.
             1. Append a new Record { [[Type]]: `"percentSign"`, [[Value]]: _percentSignSymbol_ } as the last element of _result_.
-          1. Else if _p_ is equal `"currency"` and _numberFormat_.[[Style]] is `"currency"`, then
+          1. Else if _p_ is equal `"measureUnitSymbol"` and _numberFormat_.[[UnitType]] is `"measureUnit"`, then
```

Comment:
Why are we checking for the `[[UnitType]]` here and below? I'd suggest, instead, to add assertions inside the conditional that the `[[UnitType]]` is `measureUnit` if `_p_ equals "measureUnitSymbol"` (and you can note these assertions when you're defining the locale data).